src/Encoding.chpl

```diff
@@ -0,0 +1,248 @@
+/* Documentation for Encoding */
+module Encoding {
+
+  use IO;
+
+  type writingChannel = channel(true,iokind.little,false);
+  type readingChannel = channel(false,iokind.little,false);
+
+  // wireTypes
+  const varint = 0;
+  const lengthDelimited = 2;
+  const fixed64Type = 1;
+  const fixed32Type = 5;
+
+  proc unsignedVarintAppend(val:uint, ch: writingChannel) throws {
```

Comment:
It's not something you have to change for this PR, but I wonder if `appendUnsignedVarint` would be a nicer method name. Not a big deal either way right now. (But, while these functions are not really intended to be user-facing, they do tie the protocol buffer compiler to particular versions of this module, so they should be stable). I recommend we revisit these names when we do the API review.

Reply:
Yes, agreed. I have also opened an issue for the same(https://github.com/Aniket21mathur/Chapel-protobuf/issues/12).